pkg/cache/cache.go

```diff
@@ -0,0 +1,195 @@
+package cache
+
+import (
+	"fmt"
+	"github.com/patrickmn/go-cache"
+	"k8s.io/client-go/util/workqueue"
+	//"reflect"
+	glog "github.com/Sirupsen/logrus"
+	"k8s.io/client-go/pkg/api/v1"
+	k8sCache "k8s.io/client-go/tools/cache"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	calicoClient "github.com/projectcalico/libcalico-go/lib/client"
+	"github.com/projectcalico/libcalico-go/lib/api/unversioned"
+	"github.com/kylelemons/godebug/pretty"
+)
+
+type ResourceCache interface {
+	// Sets the key to the provided value, and generates an update
+	// on the queue the value has changed.
+	Set(key string, value interface{})
+
+	// Gets the value associated with the given key.  Returns nil
+	// if the key is not present.
+	Get(key string) (interface{}, bool)
+
+	// Sets the key to the provided value, but does not generate
+	// and update on the queue ever.
+	// Priming should happen only at the begining and only run 
+	// method should have access to it. Hence a private method.
+	prime(key string, value interface{})
```

Comment:
I think this should be a Public interface - the workloadendpoint controller is going to need this.